src/main/webapp/dev/js/common/questionRubric.js

```diff
@@ -367,6 +379,7 @@ function toggleAssignWeightsRow($checkbox) {
     if ($checkbox.prop('checked')) {
         $weightsRow.show();
         $weightsRowFirstCell.html('Weights <span class="glyphicon glyphicon-arrow-right"></span>');
+        makeRubricWeightsRequired($weightsRow);
```

Comment:
Due to `removeRequiredIfElementHidden` function in `InstructorFeedbackEdit.js`, the following problem arises.

**Steps:** 
- click on `Add question` button and choose rubric question
- click on `Save question` without adding the question text (or make changes that will violate a front-end validation check)
- The question will not be added, as the question doesn't have a question text
- Enter the question text
- Now, check `Choices are weighted` checkbox
- leave a weight cell empty
- Save question

The expected behaviour is that the question will not be added as the field is `required`
The actual behaviour is that the front-end validation does not work

The reason for this behaviour is that, when we are trying to save the question without entering the text, as we are clicking the `Save Question` button, all the hidden fields that are `required` (including the weight cells as at that point they are hidden) are now not required. So, after that when we try to leave a weight cell empty, the front-end validation doesn't work as the weight attribute has been removed by the `removeRequiredIfElementHidden` method. 

SO, to solve this, I have added this method that will make all the weight cells required when the `Choices are weighted` checkbox is checked. For added safety I have added the `required` attribute in the `feedbackQuestionRubricEditFormWeightFragment.html` file and in `addRubricCol(questionNum)` method as well.